acmeproxy/proxy/tests/test_commands.py

```diff
@@ -0,0 +1,125 @@
+        )
+        out = StringIO()
+        call_command("listresponses", stdout=out)
+        assert "example.com" in out.getvalue()
+
+    def test_nothing(self):
+        out = StringIO()
+        with pytest.raises(CommandError):
+            call_command("listresponses", stdout=out)
+
+
+@pytest.mark.django_db
+class TestPipeAPI:
+    @pytest.mark.parametrize(
+        "test_input, output",
+        [
+            ("HELO\t1\nDEBUGQUIT", "OK\tACME Proxy API\n"),
+            (
+                "HELO\t1\nQ\texample.com\th\tANY\th\t127.0.0.1\nDEBUGQUIT",
+                'OK\tACME Proxy API\nDATA\texample.com\tIN\tSOA\t5\t1\tacme-proxy-ns1.example.com. hostmaster.example.com. 1592267735 0 0 0 0\nDATA\texample.com\tIN\tNS\t5\t1\tacme-proxy-ns1.example.com\nDATA\texample.com\tIN\tCAA	5\t1\t0 issue "letsencrypt.org"\nEND\n',
+            ),
+            (
+                "HELO\t1\nQ\texample.com\th\tCAA\th\t127.0.0.1\nDEBUGQUIT",
+                'OK\tACME Proxy API\nDATA\texample.com\tIN\tCAA	5\t1\t0 issue "letsencrypt.org"\nEND\n',
+            ),
+            (
+                "HELO\t1\nQ\t_acme-challenge.example.com\th\tANY\th\t127.0.0.1\nDEBUGQUIT",
+                'OK\tACME Proxy API\nDATA\t_acme-challenge.example.com	IN\tTXT	5\t1\ttest_response\nDATA\t_acme-challenge.example.com\tIN\tSOA\t5\t1\tacme-proxy-ns1.example.com. hostmaster.example.com. 1592267735 0 0 0 0\nDATA\t_acme-challenge.example.com\tIN\tNS\t5\t1\tacme-proxy-ns1.example.com\nDATA\t_acme-challenge.example.com\tIN\tCAA	5\t1\t0 issue "letsencrypt.org"\nEND\n',
+            ),
+            (
+                "HELO\t1\nQ\t_acme-challenge.example.com\th\tTXT\th\t127.0.0.1\nDEBUGQUIT",
+                "OK\tACME Proxy API\nDATA\t_acme-challenge.example.com	IN\tTXT	5\t1\ttest_response\nEND\n",
+            ),
+            ("RANDOM GARBAGE", "FAIL\n",),
+            ("RANDOMGARBAGE", "FAIL\n",),
+            ("HELO\t1\nRANDOMGARBAGE\nDEBUGQUIT", "OK\tACME Proxy API\n",),
+            (
+                "HELO\t1\nR random correct number of things\nDEBUGQUIT",
+                "OK\tACME Proxy API\nEND\n",
+            ),
+            (
+                "HELO\t1\nRANDOMGARBAGE\nQ\texample.com\th\tCAA\th\t127.0.0.1\nDEBUGQUIT",
+                'OK\tACME Proxy API\nDATA\texample.com\tIN\tCAA	5\t1\t0 issue "letsencrypt.org"\nEND\n',
+            ),
+            (
+                "HELO\t1\nR random correct number of things\nQ\texample.com\th\tCAA\th\t127.0.0.1\nDEBUGQUIT",
+                'OK\tACME Proxy API\nEND\nDATA\texample.com\tIN\tCAA	5\t1\t0 issue "letsencrypt.org"\nEND\n',
+            ),
+        ],
+    )
+    def test_query(self, monkeypatch, test_input, output):
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        Response.objects.create(
+            name="example.com", response="test_response", created_by_ip="127.0.0.1"
+        )
```

Comment:
This seems to be a common pattern being used.

If its only the name values that matter, I wonder if the tests might read a bit better with the object creation code extracted:

```suggestion
        create_authorisation(name="example.com")
        create_response(name="example.com")
```
And then:
```
def create_authorisation(name):
    Authorisation.objects.create(
        name=name, secret="test_secret", created_by_ip="127.0.0.1"
    )

def create_response(name):
    Response.objects.create(
        name=name, response="test_response", created_by_ip="127.0.0.1"
    )
```

I accept this might not really be worth the effort. I will leave it up to you.